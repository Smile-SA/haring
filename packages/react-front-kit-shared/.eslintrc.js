module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
