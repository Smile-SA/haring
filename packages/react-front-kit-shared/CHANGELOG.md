# @smile/react-front-kit-shared

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/react-front-kit/pull/53) [`01ee3ea`](https://github.com/Smile-SA/react-front-kit/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
