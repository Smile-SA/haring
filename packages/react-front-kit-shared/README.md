# React Front Kit shared

Shared components and functions.

## Documentation

You can access the documentation here: https://smile-sa.github.io/react-front-kit-doc

## Prerequisite

You must have React 18 or later installed on your project.

## Installation

Install `@smile/react-front-kit-shared` and peer dependencies with:

```bash
npm i @smile/react-front-kit-shared @mantine/core @mantine/hooks @phosphor-icons/react
```

## Styles

Import the styles in your app with:

```js
import '@mantine/core/styles.css';
import '@smile/react-front-kit-shared/style.css';
```

## License

LGPL-3.0
