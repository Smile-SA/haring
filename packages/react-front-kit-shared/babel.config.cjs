const { babelConfig } = require('test');

module.exports = babelConfig;
