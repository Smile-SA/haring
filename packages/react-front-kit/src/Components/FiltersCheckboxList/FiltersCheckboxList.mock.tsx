export const filters = [
  { active: false, id: 1, label: 'Tomato', value: 'tomato' },
  { active: false, id: 2, label: 'Banana', value: 'banana' },
  { active: true, id: 3, label: 'apple', value: 'apple' },
  { active: true, id: 4, label: 'Orange ', value: 'orange' },
];
