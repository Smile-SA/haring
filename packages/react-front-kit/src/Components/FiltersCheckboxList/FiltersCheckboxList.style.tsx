import { createStyles } from '@mantine/core';

export const useStyles = createStyles((theme) => ({
  checkboxRoot: {
    marginBottom: '10px',
  },
  checkboxsBottom: {
    borderTop: `1px solid ${theme.colors.gray[2]}`,
    padding: '17px 0 24px',
  },
  checkboxsTop: {
    margin: '16px 0',
  },
}));
