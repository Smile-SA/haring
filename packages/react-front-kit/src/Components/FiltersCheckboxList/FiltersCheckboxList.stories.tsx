import type { Meta, StoryObj } from '@storybook/react';

import { FiltersCheckboxList as Cmp } from './FiltersCheckboxList';
import { filters } from './FiltersCheckboxList.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/FiltersCheckboxList',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const FiltersCheckboxList: IStory = {
  args: {
    filters,
  },
};
