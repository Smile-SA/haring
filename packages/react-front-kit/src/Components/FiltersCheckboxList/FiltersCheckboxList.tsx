'use client';

import type { ReactElement } from 'react';

import {
  Button,
  Checkbox,
  Divider,
  Input,
  useMantineTheme,
} from '@mantine/core';
import { MagnifyingGlass } from '@phosphor-icons/react';
import { useState } from 'react';

import { useStyles } from './FiltersCheckboxList.style';

export interface IFilter {
  active: boolean;
  id: number | string;
  label: string;
  value: string;
}

export interface IFiltersCheckboxListProps {
  buttonLabel?: string;
  filters?: IFilter[];
  onClickButton?: (newFilters: IFilter[]) => void;
  placeholder?: string;
}

export function FiltersCheckboxList(
  props: IFiltersCheckboxListProps,
): ReactElement {
  const {
    buttonLabel = 'Validate changes',
    placeholder = 'Search in filters',
    filters = [],
    onClickButton,
  } = props;
  const [searchInput, setSearchInput] = useState('');
  const [newFilters, setNewFilters] = useState(filters);

  const theme = useMantineTheme();
  const { classes } = useStyles();

  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>): void {
    setSearchInput(event.target.value);
  }

  function handleCheckboxChange(item: IFilter): void {
    setNewFilters(
      newFilters.map((element) => {
        if (element.id !== item.id) {
          return element;
        }
        element.active = !element.active;
        return element;
      }),
    );
  }

  function filterIncludeSearchInput(label: string): boolean {
    return label.includes(searchInput.toLocaleLowerCase());
  }

  return (
    <>
      <Input
        onChange={handleInputChange}
        placeholder={placeholder}
        rightSection={
          <MagnifyingGlass
            color={theme.colors.cyan[9]}
            size={20}
            values={searchInput}
            weight="bold"
          />
        }
      />
      <div className={classes.checkboxsTop}>
        {filters.map((item) => {
          if (!item.active && filterIncludeSearchInput(item.label)) {
            return (
              <Checkbox
                key={item.id}
                classNames={{ root: classes.checkboxRoot }}
                defaultChecked={item.active}
                label={item.label}
                onChange={() => handleCheckboxChange(item)}
              />
            );
          }
          return '';
        })}
      </div>
      <Divider color="gray.2" />
      <div className={classes.checkboxsBottom}>
        {filters.map((item) => {
          if (item.active && filterIncludeSearchInput(item.label)) {
            return (
              <Checkbox
                key={item.id}
                classNames={{ root: classes.checkboxRoot }}
                defaultChecked={item.active}
                label={item.label}
                onChange={() => handleCheckboxChange(item)}
              />
            );
          }
          return '';
        })}
      </div>
      <Button fullWidth onClick={() => onClickButton?.(newFilters)} size="md">
        {buttonLabel}
      </Button>
    </>
  );
}
