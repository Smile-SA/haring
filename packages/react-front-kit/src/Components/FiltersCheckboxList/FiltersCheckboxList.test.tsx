import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { FiltersCheckboxList } from './FiltersCheckboxList';

describe('FiltersCheckboxList', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <FiltersCheckboxList
        buttonLabel="Test"
        filters={[]}
        placeholder="Test"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
