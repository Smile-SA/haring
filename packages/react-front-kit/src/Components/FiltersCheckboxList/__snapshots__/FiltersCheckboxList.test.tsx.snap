// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersCheckboxList matches snapshot 1`] = `
<div>
  <div
    class="mantine-Input-wrapper mantine-1v7s5f8"
  >
    <input
      aria-invalid="false"
      class="mantine-Input-input mantine-v2g2mm"
      placeholder="Test"
    />
    <div
      class="mantine-1nmvwx4 mantine-Input-rightSection"
    >
      <svg
        fill="#0b7285"
        height="20"
        values=""
        viewBox="0 0 256 256"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M232.49,215.51,185,168a92.12,92.12,0,1,0-17,17l47.53,47.54a12,12,0,0,0,17-17ZM44,112a68,68,0,1,1,68,68A68.07,68.07,0,0,1,44,112Z"
        />
      </svg>
    </div>
  </div>
  <div
    class="mantine-1o8kcxl"
  />
  <div
    class="mantine-Divider-root mantine-Divider-horizontal mantine-1rts92h"
    role="separator"
  />
  <div
    class="mantine-ggbtxo"
  />
  <button
    class="mantine-UnstyledButton-root mantine-Button-root mantine-1t8lq1h"
    data-button="true"
    type="button"
  >
    <div
      class="mantine-1wpc1xj mantine-Button-inner"
    >
      <span
        class="mantine-1ryt1ht mantine-Button-label"
      >
        Test
      </span>
    </div>
  </button>
</div>
`;
